### ISSUE TITLE
On a freeze or object that cannot place focus, focus desktop

### ISSUE THREAD

**Issue by alexstine** [nvaccess/nvda#10382]
### Is your feature request related to a problem? Please describe.
Whenever NVDA gets on a big web page or you close a Microsoft program, sometimes a crash occurs to the point where you must log off as speech is completely lost and CTRL+ALT+N does not restart NVDA.
### Describe the solution you'd like
Switch the window and focus on desktop. The desktop never goes away so instead of a straight crash, try to focus somewhere that will not be confusing for the user.
### Describe alternatives you've considered
Ensure that CTRL+ALT+N will restart NVDA in all cases. Completely kill the process and re-execute it in case of a complete failure/process lockup.
### Additional context
I am using Windows 7 Professional, 16GB RAM, Intel quad core 3.7GHZ CPU.

**Comment by Brian1Gaff** [comment]
Yes this would be nice, but how would you undo the crash to make such a 
thing work? I don't see how this can be done in most cases. Anyone?
 Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal E-mail to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
Newsgroup monitored: alt.comp.blind-users
----- Original Message -----

**Comment by leonardder** [comment]
Could you please provide steps to reproduce the crash you're experiencing?

**Comment by tech55541** [comment]
There are no specific steps. It just happens on some websites and then NVDA will not respond. Just straight freeze where even a restart doesn't work with CTRL+ALT+N. A process shouldn't crash like that and it has recently been getting worse after the last couple updates. It's like code is not optimized to deal with errors it might encounter and just gives up. In all cases a restart should work. I think that's the first place to start looking.

I will share my debug logs when I can make it happen again.

Thanks.

**Comment by tech55541** [comment]
@leonardder I finally have logs for you. As you can see, there is a nasty trace back relating to the gainFocus method.

In an event like this, why can you not focus on desktop? We already have an understanding that focus item might be invalid, so else to another one preventing the complete freeze up.

Thanks.
[nvda-output.log](https://github.com/nvaccess/nvda/files/3775311/nvda-output.log)

**Comment by leonardder** [comment]
Letting the screen reader change focus is usually not preferred. Why do you think focussing the desktop will benefit you if the keyboard input is frozen anyway?

**Comment by leonardder** [comment]
@tech55541 I'm seeing the "theems manager" add-on on your system. That reminds me of the question whether you're able to reproduce this without add-ons?

**Comment by Adriani90** [comment]
@tech55541 could you please give an update to this issue with NVDA 2019.3 Beta 2 and addons disabled? ARe the crashes still reproducible for you? I cannot experience any crashes on my side with addons disabled.

**Comment by tech55541** [comment]
Hello,

Let me see if I can find someone else to test. I am still on Windows 7 for backwards compatibility with work software, not sure if my test data would be all that relevant now. I now use Mac OSX for my day to day operations, so lost track of this.

**Comment by Adriani90** [comment]
This seems very specific to a certain machine and we do not have enough user reports on this, actually I didn't hear any further users reporting this issue sofar. So due to missing way to reproduce this, I am closing for now. But please feel free to comment if you have further details for us (i.e. test with current NVDA version 2020.1) and we can reopen in case this issue still occurs. Thanks.
